packages/astro/src/vite-plugin-astro-server/index.ts

```diff
@@ -111,6 +111,25 @@ async function handle404Response(
 	if (pathname === '/' && !pathname.startsWith(devRoot)) {
 		html = subpathNotUsedTemplate(devRoot, pathname);
 	} else {
+		// HACK: redirect without the base path for assets in publicDir
+		const redirectTo = 
```

Comment:
so as a general rule, we usually don't want to redirect in dev since the developer might accidentally rely on that behavior during dev, and then ship to prod with broken code. Not sure I 100% understand what's going on here so can't say for sure, just as long as you're accounting for the goal: "don't implement helpful behavior in dev that might not exist in prod. It's usually better to error."

Now having said that, we do default `trailingSlash` to `ignore`, so maybe developer convenience is also worth taking into account.